doc/contrib/new_api.md

```diff
@@ -0,0 +1,510 @@
+# PREVIEW: New `libelektra-core` API 2023
+
+## Core API
+
+### Note on names
+
+From the [use cases](/doc/usecases/core/README.md) we know that there is a collection structure which contains elements with a name and value (and metadata).
+
+The current API uses these names for those concepts:
+
+- Collection: `KeySet`
+- Element: `Key`
+- Name of element: —
+- Value of element: —
+
+The API proposed below uses (to keep as much as possible from above):
+
+- Collection: `ElektraKeyset`
+- Element: `ElektraKey`
+- Name of element: `ElektraKeyname`
+- Value of element: `ElektraKeyvalue`
+
+Another (maybe less confusing) option would be:
+
+- Collection: `ElektraSet` or `ElektraMap`
+- Element: `ElektraEntry`
+- Name of element: `ElektraKey` or `ElektraName`
```

Comment:
Probably `ElektraKey` if we keep KDB.

Reply:
We could go with EDB for ElektraDatabase. Would be another huge change, but no bigger than "Key -> Entry" really.

Reply:
I don't have troubles with EDB¹ and I don't think that the EDB change is as difficult, as *every* occurrence from KDB would be EDB afterwards. The trouble with Key is that there are many occurrences of Key not having the meaning of "ElektraEntry". But you probably know that very well anyway.

¹ historically there was the idea that "Elektra" is only a marketing name but not used in technical context. But I think we are already way beyond that point.